core/arch/arm/kernel/thread_optee_smc.c

```diff
@@ -636,11 +636,14 @@ static struct mobj *get_rpc_alloc_res(struct optee_msg_arg *arg,
 
 	if (!mobj) {
 		thread_rpc_free(bt, cookie, mobj);
-		return NULL;
+		goto out;
 	}
 
 	assert(mobj_is_nonsec(mobj));
-
+out:
+	if (!mobj)
+		EMSG("RPC allocation failed in non-secure world: ret=%#"PRIx32
+		     " ret_origin=%#"PRIx32, arg->ret, arg->ret_origin);
```

Comment:
The log message seems ok when `ret` reports an error but when `ret` is 0, the failure comes from another origin, either issue with the RPC ABI or out-of-mem in TEE core. We may not be able to distinguish the 2. Maybe rephrase:
```patch
-		EMSG("RPC allocation failed in non-secure world: ret=%#"PRIx32
+		EMSG("RPC allocation failed. Non-secure world result: ret=%#"PRIx32

Reply:
Agreed.